From 50c4175175e5e26cb1fc95dcce20b5d80d4f825f Mon Sep 17 00:00:00 2001
From: Faiz Abbas <faiz_abbas@ti.com>
Date: Tue, 28 Jan 2020 15:59:39 +0530
Subject: [PATCH 33/50] mmc: am654_sdhci: Add Support for 3 bit frequency
 select

The j721e 8 bit instance phy has a 3 bit frequency selection field.
Incorporate this by creating a FREQSEL_2_BIT flag which can be used to
differentiate between the freqsel fields.

Signed-off-by: Faiz Abbas <faiz_abbas@ti.com>
---
 drivers/mmc/am654_sdhci.c | 17 +++--------------
 1 file changed, 3 insertions(+), 14 deletions(-)

diff --git a/drivers/mmc/am654_sdhci.c b/drivers/mmc/am654_sdhci.c
index 730326f498..d7ae1ecf8d 100644
--- a/drivers/mmc/am654_sdhci.c
+++ b/drivers/mmc/am654_sdhci.c
@@ -82,6 +82,7 @@ struct am654_sdhci_plat {
 	u32 flags;
 #define DLL_PRESENT	(1 << 0)
 #define IOMUX_PRESENT	(1 << 1)
+#define FREQSEL_2_BIT	(1 << 2)
 	bool dll_on;
 };
 
@@ -116,7 +117,6 @@ static int am654_sdhci_set_ios_post(struct sdhci_host *host)
 	struct am654_sdhci_plat *plat = dev_get_platdata(dev);
 	unsigned int speed = host->mmc->clock;
 	int sel50, sel100, freqsel;
-	u32 otap_del_sel;
 	u32 mask, val;
 	int ret;
 
@@ -140,18 +140,7 @@ static int am654_sdhci_set_ios_post(struct sdhci_host *host)
 		otap_del_sel = plat->otap_del_sel[host->mmc->selected_mode];
 		mask = OTAPDLYENA_MASK | OTAPDLYSEL_MASK;
 		val = (1 << OTAPDLYENA_SHIFT) |
-		      (otap_del_sel << OTAPDLYSEL_SHIFT);
-
-		/* Write to STRBSEL for HS400 speed mode */
-		if (host->mmc->selected_mode == MMC_HS_400) {
-			if (plat->flags & STRBSEL_4_BIT)
-				mask |= STRBSEL_4BIT_MASK;
-			else
-				mask |= STRBSEL_8BIT_MASK;
-
-			val |= plat->strb_sel << STRBSEL_SHIFT;
-		}
-
+		      (plat->otap_del_sel << OTAPDLYSEL_SHIFT);
 		regmap_update_bits(plat->base, PHY_CTRL4, mask, val);
 
 		if (plat->flags & FREQSEL_2_BIT) {
@@ -210,7 +199,7 @@ const struct sdhci_ops am654_sdhci_ops = {
 
 const struct am654_driver_data am654_drv_data = {
 	.ops = &am654_sdhci_ops,
-	.flags = DLL_PRESENT | IOMUX_PRESENT,
+	.flags = DLL_PRESENT | IOMUX_PRESENT | FREQSEL_2_BIT,
 };
 
 const struct am654_driver_data j721e_8bit_drv_data = {
-- 
2.17.1

